e2etest/status/ipaddresspool.go

```diff
@@ -0,0 +1,234 @@
+	"go.universe.tf/metallb/e2etest/pkg/service"
+)
+
+var (
+	ConfigUpdater config.Updater
+	Reporter      *k8sreporter.KubernetesReporter
+)
+
+var _ = ginkgo.Describe("IPAddressPool status", func() {
+	var cs clientset.Interface
+	var f *framework.Framework
+
+	f = framework.NewDefaultFramework("status")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+
+		err := ConfigUpdater.Clean()
+		framework.ExpectNoError(err)
+	})
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentSpecReport().Failed() {
+			k8s.DumpInfo(Reporter, ginkgo.CurrentSpecReport().LeafNodeText)
+		}
+		err := ConfigUpdater.Clean()
+		framework.ExpectNoError(err)
+	})
+
+	ginkgo.Context("Service updates", func() {
+		ginkgo.It("performs changes on services and checks pool status", func() {
+			pool1 := "pool1"
+			pool2 := "pool2"
+
+			resources := config.Resources{
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: pool1,
+						},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{
+								"20.20.20.0/24",
+								"2000::/121",
+							},
+						},
+					},
+				},
+			}
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Creating a LB service and checking status of pool 1")
+			jig := e2eservice.NewTestJig(cs, f.Namespace.Name, "test-svc1")
+			svc1, err := jig.CreateLoadBalancerService(10*time.Second, service.DualStack)
+			framework.ExpectNoError(err)
+
+			defer func() {
+				service.Delete(cs, svc1)
+			}()
+
+			gomega.Eventually(func() metallbv1beta1.IPAddressPoolStatus {
+				p := metallbv1beta1.IPAddressPool{}
+				err := ConfigUpdater.Client().Get(context.Background(), client.ObjectKey{Name: pool1, Namespace: ConfigUpdater.Namespace()}, &p)
+				framework.ExpectNoError(err)
+				return p.Status
+			}, 5*time.Second, 1*time.Second).Should(gomega.Equal(metallbv1beta1.IPAddressPoolStatus{
+				AssignedIPv4:  1,
+				AssignedIPv6:  1,
+				AvailableIPv4: 255,
+				AvailableIPv6: 127,
+			}))
+
+			ginkgo.By("Creating a second LB service and checking status of pool 1")
+			jig = e2eservice.NewTestJig(cs, f.Namespace.Name, "test-svc2")
+			svc2, err := jig.CreateLoadBalancerService(10*time.Second, service.DualStack)
+			framework.ExpectNoError(err)
+
+			defer func() {
+				service.Delete(cs, svc2)
+			}()
+
+			gomega.Eventually(func() metallbv1beta1.IPAddressPoolStatus {
+				p := metallbv1beta1.IPAddressPool{}
+				err := ConfigUpdater.Client().Get(context.Background(), client.ObjectKey{Name: pool1, Namespace: ConfigUpdater.Namespace()}, &p)
+				framework.ExpectNoError(err)
+				return p.Status
+			}, 5*time.Second, 1*time.Second).Should(gomega.Equal(metallbv1beta1.IPAddressPoolStatus{
+				AssignedIPv4:  2,
+				AssignedIPv6:  2,
+				AvailableIPv4: 254,
+				AvailableIPv6: 126,
+			}))
+
+			ginkgo.By("Moving the second LB service to pool2 and checking status of pool 1 and 2")
```

Comment:
nit: change this to something like "Creating another pool and verifying status.." and have another By before moving the service